python/servo/command_base.py

```diff
@@ -480,6 +481,68 @@ def get_nightly_binary_path(self, nightly_date: str | None) -> str | None:
     def msvc_package_dir(self, package: str) -> str:
         return servo.platform.windows.get_dependency_dir(package)
 
+    def get_current_clang_version(self, CC: str) -> Optional[str]:
+        try:
+            result = subprocess.run(
+                [CC, "--version"],
+                check=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                text=True,
+            )
+            clang_version_full_string_split = result.stdout.lower().split()
+            for i, word in enumerate(clang_version_full_string_split):
+                if word == "version":
+                    if i + 1 < len(clang_version_full_string_split):
+                        return clang_version_full_string_split[i + 1]
+                    else:
+                        return None
```

Comment:
Maybe try this approach for iterator through pairs instead: https://stackoverflow.com/a/5764948 ?